File: solutions/elixir/secret-handshake/lib/secret_handshake.ex
defmodule SecretHandshake do
  import Bitwise
  @spec commands(code :: integer) :: list(String.t())
  def commands(code) do
    actions = ["wink", "double blink", "close your eyes", "jump"]

    result =
      actions
      |> Enum.with_index()
      |> Enum.filter(fn {_, index} -> (code &&& 1 <<< index) != 0 end)
      |> Enum.map(fn {action, _} -> action end)

    if (code &&& 0x10) != 0, do: Enum.reverse(result), else: result
  end
end

File: solutions/python/secret-handshake/secret_handshake.py
def commands(binary_str):
    actions = ["wink", "double blink", "close your eyes", "jump"]
    result = [actions[i] for i in range(4) if binary_str.zfill(5)[-(i + 1)] == "1"]
    return result[::-1] if binary_str.zfill(5)[0] == "1" else result
